Native reference guide updates

galderz opened the pull request:
Added a few more details and FAQs. Replaced heap dump section with details of upcoming GraalVM 22.2 support.

galderz committed:
Clarify /project paths in native build output

galderz committed:
Add warning about use of expert options

galderz committed:
Add FAQ entry around out of memory errors building natives

galderz committed:
Update native ref guide with upcoming heap dump capabilities

galderz committed:
Add note on perf record sampling value

zakkak commented on docs/src/main/asciidoc/native-reference.adoc:
```diff
@@ -767,6 +781,20 @@ Then, from inside the tools container we execute:
+but the more precision you get on what the application is doing.
+So, finding the right value is a balancing act.
+
+This number should generally be a prime number,
```
It would be great if we could add a reference to back this up if possible.

zakkak reviewed (approved):
LGTM, thanks @galderz 

I only added a comment about adding a reference backing up the prime-number recommendation for perf.

galderz commented:
Hmmm, I'll look up a reference but I think it's common sense.

I have a few more changes that I'd like to add, so I'll make this a draft.

galderz committed:
Add native ref guide to step to verify encrypt/decrypt works

galderz committed:
Always filter by GreetingResource on cypher queries

galderz committed:
Add a note on how to inspect native executables in runtime container

galderz committed:
Add FAQ entry on detecting performance issues in production

galderz committed:
Add FAQ on what information helps the most

galderz commented:
@zakkak I've removed the prime number recommendation. I've added a few more details based on discussions during the 2nd training. Can you please review it once more? Thx!

zakkak commented on docs/src/main/asciidoc/native-reference.adoc:
```diff
@@ -252,6 +260,15 @@ For example, we can see how the heap and text sections take most of binary:
+====
+Runtime containers produced by Quarkus to run native executables will not include the tools mentioned above.
+To explore a native executable within a runtime container,
+it's best to run the container itself and then `docker cp` the executable locally.
```
It might be good to include an example, e.g.

```suggestion
it's best to run the container itself and then `docker cp` the executable locally, e.g.:

----
docker run -i --rm --name=mytest -p 8080:8080 test/debugging-native:1.0.0-SNAPSHOT
docker cp mytest:/work/application path/on/host/
----
```

I am not sure you can nest a code block in a note though!

galderz replied:
Just tried it, yes you can.

zakkak commented on docs/src/main/asciidoc/native-reference.adoc:
```diff
@@ -1324,6 +1371,25 @@ For example, when registering types and methods for reflection access,
 
+=== I get an `OutOfMemoryError` (OOME) building native executables, what can I do?
+
+Building native executables is not only time consuming, but it also takes a fair amout of memory.
```
```suggestion
Building native executables is not only time consuming, but it also takes a fair amount of memory.
```

zakkak commented on docs/src/main/asciidoc/native-reference.adoc:
```diff
@@ -1324,6 +1371,25 @@ For example, when registering types and methods for reflection access,
+
+Building native executables is not only time consuming, but it also takes a fair amout of memory.
+For example, building a sample native Quarkus JPA application such as the Hibernate quickstart,
+will easily use 6GB to 8GB resident set size in memory.
```
```suggestion
may use 6GB to 8GB resident set size in memory.
```

zakkak reviewed (changes requested):
LGTM, apart from the typo and a couple of suggestions.

Thanks

galderz commented:
@zakkak Applied the changes you suggested. I rebased them into their corresponding commits, to maintain each independent guide change independent.

zakkak reviewed (approved):
Thanks @galderz . It's ready to be merged AFAIC

zakkak commented:
FWIW I think that thid PR, **excluding** https://github.com/quarkusio/quarkus/pull/26094/commits/f7a71d0e5f5fc820d87b6bf368416420445f2c7b, could also be backported to `2.7`.
